frontend/src/components/Collections/components/Grid/components/DatasetsGrid/index.tsx

```diff
@@ -15,6 +16,12 @@ interface Props {
   invalidateCollectionQuery: () => void;
   visibility: VISIBILITY_TYPE;
   accessType?: Collection["access_type"];
+  isRevision: boolean;
+  onUploadFile: (
+    reuploadDataset?: any,
+    datasetId?: any
```

Comment:
The ReactQuery query method is way too hard to type as a function arg

Reply:
Ah would this work?

```ts
onUploadFile: (
    reuploadDataset?: Props["reuploadDataset"],
    datasetId?: string
  ) => ChooserProps["onUploadFile"];
```

Reply:
That leads to typing areas when the `DatasetGrid` is called in `DatasetTab`

Reply:
Updated typings: https://github.com/chanzuckerberg/corpora-data-portal/compare/seve/1060-dataset-reupload...thuang-1060-reupload?expand=1

I looked into this a little more, so it seems like TypeScript caught inconsistent return types for `uploadLink` and `reuploadDataset`. The former returns `datasetId: string` but latter returns `void`

```ts
const [uploadLink] = useCollectionUploadLinks(collectionId, visibility);
const [reuploadDataset] = useReuploadDataset(collectionId);
```

So in `frontend/src/views/Collection/components/DatasetTab/index.tsx`'s `addNewFile()`, config.onSuccess will either get `string` or `void`, but our callback function says it will always be a string, which is untrue, thus the inconsistency that TypeScript is complaining about?

```ts
const payload = JSON.stringify({ url: newFile.link, uuid: originalId });
      mutationFunction(
        { collectionId: collectionId, payload },
        {
          onSuccess: (datasetID: Dataset["id"]) => {
            newFile.id = datasetID;
            DatasetUploadToast.show({
              icon: IconNames.TICK,
              intent: Intent.PRIMARY,
              message:
                "Your file is being uploaded which will continue in the background, even if you close this window.",
            });
            setUploadedFiles({ ...uploadedFiles, [newFile.id]: newFile });
            queryCache.invalidateQueries(USE_COLLECTION);
          },
        }
      );
```